FEXCore/Source/Interface/Core/CPUBackend.h

```diff
@@ -144,31 +177,37 @@ namespace CPU {
 
     bool IsAddressInCodeBuffer(uintptr_t Address) const;
 
+    fextl::shared_ptr<CodeBuffer> GetThreadLocalCodeBuffer() const;
+
+    // Returns true if the active CodeBuffer is shared with any other threads
+    bool UsesSharedCodeBuffer() const;
+
+    // Updates the CodeBuffer if needed and returns a reference to the old one.
+    // The returned reference should be kept alive carefully to avoid early deletion of resources.
+    [[nodiscard]]
+    fextl::shared_ptr<CodeBuffer> CheckCodeBufferUpdate();
+
   protected:
     // Max spill slot size in bytes. We need at most 32 bytes
     // to be able to handle a 256-bit vector store to a slot.
     constexpr static uint32_t MaxSpillSlotSize = 32;
 
     FEXCore::Core::InternalThreadState* ThreadState;
 
-    size_t InitialCodeSize, MaxCodeSize;
+    size_t MaxCodeSize;
     [[nodiscard]]
     CodeBuffer* GetEmptyCodeBuffer();
 
-    // This is the current code buffer that we are tracking
-    CodeBuffer* CurrentCodeBuffer {};
+    // This is the code buffer actively used by this thread
+    fextl::shared_ptr<CodeBuffer> CurrentCodeBuffer;
```

Comment:
Maybe rename to ActiveCodeBuffer

Reply:
I think CurrentCodeBuffer is more fitting. `Active` has a ring of "being dynamic" to it, which would be misleading here: The current CodeBuffer may be read-only and hence will need to be updated when new code is compiled.

This is a weak opinion though, if you're convinced "active" is better I can change it.

Reply:
I'll update the comment though: `// This is the code buffer containing the main code under execution by this thread`